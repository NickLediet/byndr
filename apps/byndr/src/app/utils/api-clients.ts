import axios from 'axios';

export const NEST_API_URL = 'http://localhost:3001/api/v1';

export const cardsApiClient = axios.create({
    baseURL: NEST_API_URL,
    headers: {
        'Content-Type': 'application/json',
    },
});
