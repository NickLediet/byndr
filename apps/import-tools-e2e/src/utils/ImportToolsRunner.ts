import { exec } from  'promisify-child-process'
import { join } from 'path';

export class ImportToolsRunner {
  private cliPath: string

  constructor() {
    this.cliPath = join(process.cwd(), '../../dist/apps/import-tools/main.js')
  }

   async executeCommand(command: string) {
    return exec(`node ${this.cliPath} ${command} --env ../../.env.test`)
  }
}
 