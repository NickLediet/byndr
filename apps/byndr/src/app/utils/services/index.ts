import { cardsApiClient } from "../api-clients"

export type ImportCsvServiceParams = {
    fileData: File
}

export const importCsvService = async ({ fileData }: ImportCsvServiceParams) => {
    const formData = new FormData()
    formData.append('file', fileData)
        
    return await cardsApiClient.postForm('/import', formData)
}