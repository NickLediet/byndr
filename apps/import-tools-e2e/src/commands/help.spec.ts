import { ImportToolsRunner } from '../utils';
import { join } from 'path';

describe('`byndr --help` or `byndr help`', () => {
    let importToolsRunner: ImportToolsRunner|null = null

    beforeEach(() => {
        importToolsRunner = new ImportToolsRunner(
            join(process.cwd(), '../../dist/apps/import-tools')
        )
    })

    it('matches the help-text snapshot', async () => {
      try {
        const { stdout: flagStdout } = await importToolsRunner.executeCommand('--help')
        const { stdout: commandStdout } = await importToolsRunner.executeCommand('help')
        expect(flagStdout).toMatchSnapshot()
        expect(commandStdout).toMatchSnapshot()
        expect (flagStdout).toEqual(commandStdout)
      } catch (error) {
        console.error(error);
        expect(true).toBe(false);
      }
    })
  })