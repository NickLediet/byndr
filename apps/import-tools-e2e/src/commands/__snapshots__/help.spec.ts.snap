// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`byndr --help\` or \`byndr help\` matches the help-text snapshot 1`] = `
"Usage: byndr [options] [command]

Import tools for Byndr

Options:
  -V, --version                       output the version number
  -h, --help                          display help for command

Commands:
  debug [options]                     Debugging command for testing, will be
                                      removed in production
  list:create [options]               Create a new list in Byndr
  list:import [options] <sources...>  Import data into Byndr
  list:show [options]                 Show all lists in Byndr
  list:update [options] <sources...>  Update a list in Byndr
  help [command]                      display help for command
"
`;

exports[`\`byndr --help\` or \`byndr help\` matches the help-text snapshot 2`] = `
"Usage: byndr [options] [command]

Import tools for Byndr

Options:
  -V, --version                       output the version number
  -h, --help                          display help for command

Commands:
  debug [options]                     Debugging command for testing, will be
                                      removed in production
  list:create [options]               Create a new list in Byndr
  list:import [options] <sources...>  Import data into Byndr
  list:show [options]                 Show all lists in Byndr
  list:update [options] <sources...>  Update a list in Byndr
  help [command]                      display help for command
"
`;
